File: Algorithms/Java/1502-can_make_arithmetic_progression_from_sequence.java
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        
        Arrays.sort(arr);
        int diff = arr[1] - arr[0];

        for(int i = 1; i < arr.length && i + 1 < arr.length; i++){
            if(arr[i + 1] - arr[i] != diff){
                return false;
            }
        }

        return true;



    }
}

File: Algorithms/python/1502-can_make_arithmetic_progression_from_sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        i = 0
        j = i + 1
        prev = None        
        while j < len(arr):
            if prev == None:
                prev = arr[i] - arr[j]
            elif prev != arr[i] - arr[j]:
                return False
            
            i += 1
            j += 1
        
        return True
            
        